src/reference/asciidoc/kafka.adoc

```diff
@@ -126,9 +126,57 @@ Note, however that flushing will likely significantly reduce performance.
 
 ==== Receiving Messages
 
-Messages can be received by configuring a `MessageListenerContainer` and providing a `MessageListener`, or by
+Messages can be received by configuring a `MessageListenerContainer` and providing a <<message-listener, Message Listener>>, or by
 using the `@KafkaListener` annotation.
 
+[[message-listeners]]
+===== Message Listeners
+
+When using a <<message-listener-container, Message Listener Container>> you must provide a listener to receive data.
+There are currently four supported interfaces for message listeners:
+
+[source, java]
+----
+public interface MessageListener<K, V> {} <1>
```

Comment:
That call-out link feature looks really cool 😄 !

OK. that's all for me.
Hope the next round with `batchErrorHandling` will be the last one to merge.